carma_cooperative_perception/test/test_msg_conversion.cpp

```diff
@@ -0,0 +1,119 @@
+  object_data.detected_object_common_data.detected_id.object_id = 0xBEEF;
+  object_data.detected_object_common_data.measurement_time.measurement_time_offset = -0.1;  // s
+
+  object_data.detected_object_common_data.heading.heading = 34;  // true heading; degrees
+  object_data.detected_object_common_data.obj_type.object_type =
+    object_data.detected_object_common_data.obj_type.VEHICLE;
+
+  object_data.detected_object_common_data.pos.offset_x.object_distance = 100.0;  // m
+  object_data.detected_object_common_data.pos.offset_y.object_distance = 100.0;  // m
+
+  object_data.detected_object_common_data.pos.presence_vector |=
+    object_data.detected_object_common_data.pos.HAS_OFFSET_Z;
+  object_data.detected_object_common_data.pos.offset_z.object_distance = 100.0;  // m
+
+  object_data.detected_object_common_data.speed.speed = 10;    // m/s
+  object_data.detected_object_common_data.speed_z.speed = 20;  // m/s
+
+  object_data.detected_object_common_data.accel_4_way.longitudinal = 0.5;  // m/s^2
+  object_data.detected_object_common_data.accel_4_way.lateral = 1.00;      // m/s^2
+  object_data.detected_object_common_data.accel_4_way.vert = 23.536;       // m/s^2
+  object_data.detected_object_common_data.accel_4_way.yaw_rate = 5.0;      // degrees/s
+
+  sdsm_msg.objects.detected_object_data.push_back(object_data);
+
+  const auto detection_list{carma_cooperative_perception::to_detection_list_msg(sdsm_msg)};
+  ASSERT_EQ(std::size(detection_list.detections), 1U);
+
+  const auto detection{detection_list.detections.at(0)};
+
+  EXPECT_EQ(detection.header.stamp.sec, 0);
+  EXPECT_NEAR(detection.header.stamp.nanosec, 900'000'000U, 2);  // +/- 2 ns is probably good enough
+  EXPECT_EQ(detection.header.frame_id, "15N");
+
+  EXPECT_NEAR(detection.pose.pose.position.x, 715068.54 + 100.0, 1e-2);   // m (ref pos + offset)
+  EXPECT_NEAR(detection.pose.pose.position.y, 3631576.38 + 100.0, 1e-2);  // m (ref pos + offset)
+  EXPECT_NEAR(detection.pose.pose.position.z, 300.0 + 100.0, 1e-3);       // m (ref pos + offset)
+  EXPECT_DOUBLE_EQ(detection.pose.pose.orientation.x, 0.0);
+  EXPECT_DOUBLE_EQ(detection.pose.pose.orientation.y, 0.0);
+  EXPECT_NEAR(detection.pose.pose.orientation.z, 0.479035, 1e-5);
+  EXPECT_NEAR(detection.pose.pose.orientation.w, 0.877796, 1e-5);
+
+  EXPECT_DOUBLE_EQ(detection.twist.twist.linear.x, 10.0);
+  EXPECT_DOUBLE_EQ(detection.twist.twist.linear.y, 0.0);
+  EXPECT_DOUBLE_EQ(detection.twist.twist.linear.z, 20.0);
+  EXPECT_DOUBLE_EQ(detection.twist.twist.angular.z, 5.0);
+
+  EXPECT_DOUBLE_EQ(detection.accel.accel.linear.x, 0.5);
+  EXPECT_DOUBLE_EQ(detection.accel.accel.linear.y, 1.0);
+  EXPECT_NEAR(detection.accel.accel.linear.z, 2.4 * 9.80665, 1e-4);
+
+  EXPECT_EQ(detection.id, std::to_string(0xBEEF));
+  EXPECT_EQ(detection.motion_model, detection.MOTION_MODEL_CTRV);
+}
+
+TEST(CalcDetectionTimeStamp, Simple) {}
+
+TEST(ToPositionMsg, Simple) {}
+
+TEST(ToDetectionListMsg, Simple) {}
```

Comment:
Are we able to implement these last three tests as part of this PR?

Reply:
Already done.